program/shell/errors_to_json.go

```diff
@@ -0,0 +1,47 @@
+package shell
+
+import (
+	"encoding/json"
+
+	"github.com/elves/elvish/eval"
+	"github.com/elves/elvish/parse"
+)
+
+// ErrorInJSON is the data structure for self-defined error
+type ErrorInJSON struct {
```

Comment:
This type does not need to be exported.

Suggest naming it `errorWithContextInJSON`.

The doc comment is too vague; suggested alternative:

```go
// An auxiliary struct for converting errors with diagnostics information to JSON.
```

Reply:
will do